src/NHSD.BuyingCatalogue.Ordering.Api/Controllers/ServiceRecipientsSectionController.cs

```diff
@@ -21,76 +19,28 @@ namespace NHSD.BuyingCatalogue.Ordering.Api.Controllers
             this.serviceRecipientService = serviceRecipientService ?? throw new ArgumentNullException(nameof(serviceRecipientService));
         }
 
         [HttpGet]
         public async Task<ActionResult<ServiceRecipientsModel>> GetAllAsync(CallOffId callOffId)
         {
-            if (!await context.Order.AnyAsync(o => o.Id == callOffId.Id))
-                return NotFound();
-
-            var selectedRecipients = await context.Order
-                .Where(o => o.Id == callOffId.Id)
-                .SelectMany(o => o.SelectedServiceRecipients)
-                .OrderBy(r => r.Recipient.Name)
-                .Select(r => new ServiceRecipientModel { Name = r.Recipient.Name, OdsCode = r.Recipient.OdsCode })
-                .AsNoTracking()
-                .ToListAsync();
-
-            if (selectedRecipients.Count == 0)
-            {
-                selectedRecipients = await context.Order
-                    .Where(o => o.Id == callOffId.Id)
-                    .SelectMany(o => o.OrderItems)
-                    .Where(o => o.CatalogueItem.CatalogueItemType == CatalogueItemType.Solution)
-                    .SelectMany(o => o.OrderItemRecipients)
-                    .Select(r => new ServiceRecipientModel { Name = r.Recipient.Name, OdsCode = r.Recipient.OdsCode })
-                    .Distinct()
-                    .OrderBy(r => r.Name)
-                    .AsNoTracking()
-                    .ToListAsync();
-            }
+            var selectedRecipients = await serviceRecipientService.GetAllOrderItemRecipient(callOffId);
 
-            return new ServiceRecipientsModel { ServiceRecipients = selectedRecipients };
-        }
-
-        [HttpPut]
-        [Authorize(Policy = PolicyName.CanManageOrders)]
-        public async Task<ActionResult> UpdateAsync(CallOffId callOffId, ServiceRecipientsModel model)
-        {
-            if (model is null)
-                throw new ArgumentNullException(nameof(model));
-
-            var order = await context.Order
-                .Where(o => o.Id == callOffId.Id)
-                .Include(o => o.Progress)
-                .Include(o => o.SelectedServiceRecipients)
-                .SingleOrDefaultAsync();
-
-            if (order is null)
+            if (selectedRecipients is null)
                 return NotFound();
 
-            var recipients = await serviceRecipientService.AddOrUpdateServiceRecipients(model.ServiceRecipients);
-
-            var serviceRecipients = model.ServiceRecipients.Select(r => new SelectedServiceRecipient
-            {
-                Recipient = recipients[r.OdsCode],
-            }).ToList();
-
-            order.SetSelectedServiceRecipients(serviceRecipients);
-
-            await context.SaveChangesAsync();
+            var selectedRecipientsModel = selectedRecipients.Select(r =>
+                    new ServiceRecipientModel { Name = r.Name, OdsCode = r.OdsCode })
+                    .Distinct()
+                    .ToList();
```

Comment:
Indent should just be one tab here (for consistency).

Also the `.Distinct()` can be done in the EF query in the service (see my other comment).

Reply:
Updated Indentation and removed .Distinct() from controller layer.